runtime/interpreter/value.go

```diff
@@ -17782,6 +17783,68 @@ func (v *CompositeValue) RemoveTypeKey(
 	return v.RemoveMember(interpreter, locationRange, attachmentMemberName(attachmentType))
 }
 
+func (v *CompositeValue) Iterator(interpreter *Interpreter) ValueIterator {
+	staticType := v.StaticType(interpreter)
+
+	switch typ := staticType.(type) {
+	case InclusiveRangeStaticType:
+		return NewInclusiveRangeIterator(v, interpreter, typ)
+
+	default:
+		// Must be caught in the checker.
+		panic(errors.NewUnreachableError())
+	}
+}
+
+type InclusiveRangeIterator struct {
+	rangeValue   *CompositeValue
+	next         IntegerValue
+	stepNegative bool
+}
+
+var _ ValueIterator = &InclusiveRangeIterator{}
+
+func NewInclusiveRangeIterator(
+	v *CompositeValue,
+	interpreter *Interpreter,
```

Comment:
Just a nit-pcik: Could you please swap these two params, and have the interpreter as the first param. That's the convention we have followed elsewhere.

```suggestion
	interpreter *Interpreter,
	v *CompositeValue,

```

Reply:
cool, updated `getFieldAsIntegerValue` to follow this convention as well.